frontend/src/components/UpdateThread/UpdateThreadForm.js

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import { Form, Modal, Divider } from 'antd';
+import { useNavigate } from 'react-router-dom';
+import CreateThreadContent from '../CreateThread/CreateThreadContent'
+import '../../style/CreateThread/CreateThreadForm.css'
+import CreateThreadPetInfo from '../CreateThread/CreateThreadPetInfo'
+import useThreadTypeKeywordSwitch from '../CreateThread/useThreadTypeKeywordSwitch'
+import { useDispatch, useSelector} from 'react-redux';
+import { updateThreadAsync, getThreadAsync} from '../../thunk/threadThunk';
+import moment from 'moment';
+
+function UpdateThreadForm ({ open, onUpdate, onCancel, threadId }) {
+  const [threadType, updateThreadType] = useState('');
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const user = useSelector(state => state.user);
+  const [form] = Form.useForm();
+
+  useEffect(() => {
+    dispatch(getThreadAsync(threadId)).then((res) => {
+      console.log("Got thread data: ", res);
+      const thread = res.payload.thread;
+      console.log('thread from update: ', thread);
+      // if (!thread.pet || !thread.kind) return; // some checks, shouldn't be triggered at all
+
+      axios
```

Comment:
can we use async/await for this?

Reply:
@ddm1293 I don't think there would be any difference functionally between .then() and await.

Reply:
for code consistency.